pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/saxon/AttributeNode.java

```diff
@@ -55,7 +59,17 @@ public String getURI() {
     @Override
     public Value atomize() {
         if (value == null) {
-            value = SaxonXPathRuleQuery.getAtomicRepresentation(attribute.getValue());
+            Object data = attribute.getValue();
+            if (data instanceof List) {
+                final List<?> dataList = (List<?>) data;
```

Comment:
The logic of this method is duplicated from `SaxonXPathRuleQuery::getRepresentation`. In case the list is empty, this should be using `EmptySequence.getInstance()`, which the SaxonXPathRuleQuery already does

Reply:
Fixed.